### PR TITLE
Enable `use_super_parameters` lint

### PR DESCRIPTION
**Pull request by goderbauer**
https://github.com/flutter/flutter/issues/100575

To be submitted after:
* https://github.com/flutter/flutter/pull/100905

### CODE REVIEW

**Comment by Hixie** [comment]
test-exempt: is a test

**Review by Piinks — APPROVED** [review]
LGTM!

File: packages/flutter_tools/lib/src/debug_adapters/flutter_test_adapter.dart [hunk 1]
```diff
@@ -20,6 +20,8 @@ import 'mixins.dart';
 /// A DAP Debug Adapter for running and debugging Flutter tests.
 class FlutterTestDebugAdapter extends DartDebugAdapter<FlutterLaunchRequestArguments, FlutterAttachRequestArguments>
     with PidTracker, TestAdapter {
+  // TODO(goderbauer): Figure out why super parameters in this class break internal google tests.
```
Comment (Piinks) [inline packages/flutter_tools/lib/src/debug_adapters/flutter_test_adapter.dart:23]: Is there an open issue for this that it can reference?

Comment (goderbauer) [reply packages/flutter_tools/lib/src/debug_adapters/flutter_test_adapter.dart:23]: There is! Let me find and add it.

**Pull request merged**